test/ResidualTest.cc

```diff
@@ -0,0 +1,440 @@
+/*
+ * Copyright(c) 2019 Netflix, Inc.
+ * SPDX - License - Identifier: BSD - 2 - Clause - Patent
+ */
+
+/******************************************************************************
+ * @file ResidualTest.cc
+ *
+ * @brief Unit test for Residual functions:
+ * - ResidualKernel_avx2
+ * - residual_kernel16bit_sse2_intrin
+ * - residual_kernel_sub_sampled{w}x{h}_sse_intrin
+ * - sum_residual8bit_avx2_intrin
+ *
+ * @author Cidana-Ivy
+ *
+ ******************************************************************************/
+#include <math.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <limits.h>
+#include <new>
+// workaround to eliminate the compiling warning on linux
+// The macro will conflict with definition in gtest.h
+#ifdef __USE_GNU
+#undef __USE_GNU  // defined in EbThreads.h
+#endif
+#ifdef _GNU_SOURCE
+#undef _GNU_SOURCE  // defined in EbThreads.h
+#endif
+
+#include "EbDefinitions.h"
+#include "EbPictureOperators.h"
+#include "EbIntraPrediction.h"
+#include "EbUtility.h"
+#include "EbUnitTestUtility.h"
+#include "aom_dsp_rtcd.h"
+#include "random.h"
+#include "util.h"
+
+using svt_av1_test_tool::SVTRandom;  // to generate the random
+
+namespace {
+typedef enum { REF_MIN, REF_MAX, REF_RANDOM } TestPattern;
+TestPattern TEST_PATTERNS[] = {REF_MIN, REF_MAX, REF_RANDOM};
+
+typedef std::tuple<uint32_t, uint32_t> AreaSize;
+AreaSize TEST_AREA_SIZES[] = {
+    AreaSize(4, 4),    AreaSize(4, 8),   AreaSize(8, 4),   AreaSize(8, 8),
+    AreaSize(16, 16),  AreaSize(4, 16),  AreaSize(16, 4),  AreaSize(16, 8),
+    AreaSize(8, 16),   AreaSize(32, 32), AreaSize(32, 8),  AreaSize(16, 32),
+    AreaSize(8, 32),   AreaSize(32, 16), AreaSize(16, 64), AreaSize(64, 16),
+    AreaSize(64, 64),  AreaSize(64, 32), AreaSize(32, 64), AreaSize(128, 128),
+    AreaSize(64, 128), AreaSize(128, 64)};
+
+typedef void (*SUB_SAMPLED_TEST_FUNC)(uint8_t *input, uint32_t inputStride,
+                                      uint8_t *pred, uint32_t predStride,
+                                      int16_t *residual,
+                                      uint32_t residualStride,
+                                      uint32_t areaWidth, uint32_t areaHeight,
+                                      uint8_t lastLine);
+
+typedef struct SubSampledParam {
+    AreaSize areaSize;
+    SUB_SAMPLED_TEST_FUNC subSampledTestFunc;
+} SubSampledParam;
+SubSampledParam subsampledParam[] = {
```

Comment:
Use CamelCase for types and under_score for variable names (TypeName my_instance;)